tao_compiler/mlir/disc/transforms/mhlo_placer.cc

```diff
@@ -35,6 +35,17 @@ using placement_utils::PlacementType;
 namespace disc_ral {
 namespace {
 
+PlacementType toPlacementType(StringRef s, bool default_is_gpu = false) {
+  // h -> host (CPU)
+  // d -> device (e.g. GPU)
+  // x -> xpu (same as the default device. x will be CPU is default is CPU,
+  // otherwise is GPU) s -> shape operand, usually placed on CPU as well.
+  if (s == "h" || s == "s") return PlacementType::CPU;
+  if (s == "d") return PlacementType::GPU;
+  assert(s == "x");
```

Comment:
nit: assert is ignored in release build.

Reply:
yeah，we use the same strategy as LLVM.